backend/queues/etlProcessor.js

```diff
@@ -0,0 +1,54 @@
+const { EtlJobProcessor } = require('../etl/processor')
+const { getLogger } = require('../utils/logger')
+const { Sentry } = require('../sentry')
+
+const queues = require('./queues')
+
+const logger = getLogger('queues.etlProcessor')
+
+const attachToQueue = () => {
+  const queue = queues['etlQueue']
+  queue.process(processor)
+  queue.resume() // Start if paused
+}
+
+const processor = async (job) => {
+  const queueLog = (progress, str) => {
+    logger.debug(`[${progress}%] ${str}`)
+    job.log(str)
+    job.progress(progress)
+  }
+
+  try {
+    queueLog(0, 'Starting ETL job...')
+
+    const etl = new EtlJobProcessor()
+    await etl.run(job.data)
+
+    queueLog(100, 'Finished')
+  } catch (e) {
+    // Log the exception and rethrow.
+    Sentry.captureException(e)
+    logger.error(`ETL failed:`, e)
+    throw e
+  }
+}
+
+/**
+ * Adds a recurring task to the queue for running the ETL job.
+ * Note: this only needs to be called once when setting up the ETL pipeline.
+ *
+ * @returns {Promise<string>} The repeatable job id.
+ */
+async function scheduleEtlJob() {
```

Comment:
Just curious, when/where do we call this function? What happens in case of a restart of the instance? Will we have to reschedule it? Or does bull stores scheduled jobs to redis too?

Reply:
I'll call this function manually once on staging and prod to setup the job.
That will get persisted by Bull in Redis so we won't need to call it again, unless we wipe out Redis.